drivers/include/wiegand.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2019 Hackerspace San Salvador
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_wiegand Wiegand
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ *
+ * @brief       GPIO based Wiegand driver
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Wiegand Protocol
+ *
+ * @author      Mario Gomez <mario.gomez@teubi.co>
+ */
+
+#ifndef WIEGAND_H
+#define WIEGAND_H
+
+#include <stdint.h>
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+  #endif
+
+  /**
+   * @brief   Parameters needed for device initialization
+   */
+  typedef struct {
+    gpio_t       d0;      /**< Wiegand D0 data line */
+    gpio_t       d1;      /**< Wiegand D1 data line */
+    gpio_flank_t flank;   /**< flank detection */
+  } wiegand_params_t;
+
+  /**
+   * @brief   Device descriptor for a wiegand device
+   */
+  typedef struct {
+    uint32_t code;                         /**< Wiegand code */
+    int16_t type;                          /**< Wiegand type */
+    volatile uint32_t wg_card_temp_high;   /**< Card MSB */
+    volatile uint32_t wg_card_temp;        /**< Card LSV  */
```

Comment:
Have you considered just using `uint64_t` here?

Reply:
Yes. We can use only one 64 bit integer. I suppose that the original author of the algorithm wanted to make the code a little bit more readable when decoding the Mifare 34 format.

Let me do this little change.

Reply:
I decided to leave it as is. Indeed the reason was to add a little bit of readability as not all cards are longer than 32 bits.